### DISCUSSION TITLE
Difference between instance prompt and instance token? And [filewords] questions.

### DISCUSSION THREAD

**Discussion by sk8erchen** [d8ahazard/sd_dreambooth_extension#1154] [Q&A] [answered]
Hi, when Im about to train a dreambooth model, Im always confused about the difference of the concepts of instance prompt and instance token, can someone explain or maybe give some example? And Im also confused about the explanations in the text box shown in the screenshot down below, need help.

![Snipaste_2023-04-01_00-11-57](https://user-images.githubusercontent.com/75480055/229176291-11049fa0-0762-4c9f-a508-85d43770e414.png)

As in the red boxes 1 and 2, I dont understand what that sentence means, how the [filewords] is going to be used? Shall I include the [ ] or not? Please help, thanks!

**Comment by terrificdm** [comment, marked as answer]
Check out below 2 threads and you will understand.
[thread1](https://github.com/d8ahazard/sd_dreambooth_extension/discussions/667#discussioncomment-4574896) 
[thread2](https://github.com/d8ahazard/sd_dreambooth_extension/discussions/443) 
Also, if you can read codes, check [this](https://github.com/d8ahazard/sd_dreambooth_extension/blob/main/dreambooth/utils/text_utils.py#L37) out. It can help you fully understand function of instance token and class token.

Basically speaking, if your images are without captions and you don't want to tag them, you can ignore instance/class tokens, just use instance/class prompt for your training, and training outputs might be more flexible and editable. If you do want to use filewords(caption for each image) for your training in order to gain more consistent and less editable for outputs, you may want to use two tokens to replace/insert keywords for your captions.
